regen-network/regen-web
# feat: add sell order flow

## Description

Closes: regen-network/regen-registry#996

This PR adds the create sell order action on MyEcoCredits page.

![sell action](https://user-images.githubusercontent.com/530644/177498494-6a99b6b9-1a46-4434-ba3d-46a46bda9d68.png)

New files:
- [CreateSellOrderForm.tsx](https://github.com/regen-network/regen-web/pull/1029/files#diff-003147cbe28bd11add401a91f8fd3a846a5aaa7a770fc596781b5a844b6a842b)
- [CreateSellOrderModal.tsx](https://github.com/regen-network/regen-web/pull/1029/files#diff-018f0e05bc333f01e057ef6e02c4e287e48fd51e6da7ef31e92a9e3017ec9264)
- [useCreateSellOrderSubmit.tsx](https://github.com/regen-network/regen-web/pull/1029/files#diff-a8b95b66af2d081684249f9313261c2bb63a6968cb2d65cff789ad61b8350766)

Main changes:
- [MyEcoCredits](https://github.com/regen-network/regen-web/pull/1029/files#diff-a6a5c701de2d849850de5c38ff7f1360512ff475f35eeb41f5ea6e8ba03740ac)

Also, add a new variant for `ContainedButton`, made a few changes to input components and `TxSuccessfulModal`.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...


#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1. Go to the [dashboard](https://deploy-preview-1029--regen-registry.netlify.app/ecocredits/dashboard)
2. Try out the sell action
3. You won't be able to see the confirm modal since it requires the actual transaction to be shown.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...
